### ISSUE TITLE
TransactionService.java , UnsupportedOperationException Exception

### ISSUE THREAD

**Issue by venusdharan** [moezbhatti/qksms#567]
In the file TransactionService.java, UnsupportedOperationException Exception occurs in the following method

protected void endMmsConnectivity() {
        try {
            if (LOCAL_LOGV) Log.v(TAG, "endMmsConnectivity");

```
        // cancel timer for renewal of lease
        mServiceHandler.removeMessages(EVENT_CONTINUE_MMS_CONNECTIVITY);
        if (mConnMgr != null) {
            mConnMgr.stopUsingNetworkFeature(
                    ConnectivityManager.TYPE_MOBILE,
                    "enableMMS");
        }
    }
        catch(UnsupportedOperationException hh)
        {

        }
    finally {
        releaseWakeLock();
    }
}
```

mConnMgr.stopUsingNetworkFeature() is depricated
as per android docs
https://developer.android.com/reference/android/net/ConnectivityManager.html

i Managed to workaround using catch the UnsupportedOperationException, but didn't handled the exception.
My doubts is how to properly handle this exception.

**Comment by deplacer** [comment]
what is is your targetSdkVersion?

**Comment by venusdharan** [comment]
targetSdkVersion 24

**Comment by deplacer** [comment]
as stated in the documentary

> 	stopUsingNetworkFeature(int networkType, String feature)
This method was deprecated in API level 21. Deprecated in favor of the cleaner unregisterNetworkCallback(NetworkCallback) API. In M, and above, this method is unsupported and will throw UnsupportedOperationException if called.

that method will throw UnsupportedOperationException on API 23 and above..
reverting the version will fixed the problem.

**Comment by StephaneBausseron** [comment]
From API 22 to API 24, many others problems occurs (mainly due to runtime permission management).

@venusdharan You works on android 24 support ?

**Comment by venusdharan** [comment]
Yes, i managed to make it work by catching the UnsupportedOperationException in a variable but i couldn't figure how to handle the exception. This is what i am trying to figure.

**Comment by StephaneBausseron** [comment]
This issue is blocking for me too.
I have a branch for Android N support with Runtime Permission. I need to rewrite this method with the new API.

**Comment by moezbhatti** [comment]
Runtime permissions are on my to-do list for the next major release

**Comment by moezbhatti** [comment]
Should be fixed in QKSMS 3.0
